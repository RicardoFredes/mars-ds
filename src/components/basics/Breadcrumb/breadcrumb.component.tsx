import type { DropdownMenuItemProps } from "../DropdownMenuItem";
import type { BreadcrumbProps } from "./breadcrumb.types";

import classNames from "classnames";

import Text from "../../typographies/Text";
import ToggleDropdown from "../ToggleDropdown";

const Breadcrumb = ({ className, list, componentLink, ...props }: BreadcrumbProps) => {
  if (!Array.isArray(list)) return null;

  const cn = classNames("breadcrumb", className);
  const arialLabel = "breadcrumbs";

  if (list.length <= 3) {
    return (
      <nav className={cn} {...props} aria-label={arialLabel}>
        <ol>
          {list.map((item, index) => (
            <BreadcrumbListItem key={index} componentLink={componentLink} {...item} />
          ))}
        </ol>
      </nav>
    );
  }

  const items = [...list];
  const firstItem = items.shift();
  const lastItem = items.pop();
  const toggleButton = { iconName: "more-horizontal" };

  return (
    <nav className={cn} {...props} aria-label={arialLabel}>
      <ol>
        <BreadcrumbListItem componentLink={componentLink} {...firstItem} />
        <li>
          <ToggleDropdown componentLink={componentLink} list={items} toggleButton={toggleButton} />
        </li>
        <BreadcrumbListItem componentLink={componentLink} {...lastItem} />
      </ol>
    </nav>
  );
};

const BreadcrumbListItem = ({
  componentLink,
  label,
  className,
  ...props
}: DropdownMenuItemProps) => {
  const cn = classNames("breadcrumb__item", className);
  const Component = componentLink || "a";
  return (
    <li data-testid="breadcrumb-item" className={cn}>
      <Component {...props}>
        <Text as="span">{label}</Text>
      </Component>
    </li>
  );
};

export default Breadcrumb;
