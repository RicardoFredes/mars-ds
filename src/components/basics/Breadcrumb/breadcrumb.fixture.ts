export const mountListMock = (list: string[]) =>
  list.map((label, index) => ({
    label,
    href: `#item-${index}`,
    onClick: (event: { preventDefault: () => void }) => {
      event.preventDefault();
      console.log(label);
    },
  }));

export const mockDefaultBreadcrumbs = mountListMock([
  "ENEM",
  "Matemática",
  "Números Primos e Decomposição",
]);

export const mockBreadcrumbListExpand = mountListMock([
  "ENEM",
  "Matérias",
  "Matemática",
  "Introdução à Matemática",
  "Números Primos e Decomposição",
]);
