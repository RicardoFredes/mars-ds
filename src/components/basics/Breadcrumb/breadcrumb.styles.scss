.breadcrumb {
  ol {
    list-style: none;
    padding: 0;
    margin: 0;

    display: flex;
    align-items: center;
    min-height: 40px;
    white-space: nowrap;

    > li {
      display: flex;
      align-items: center;
      > a {
        $color: $color-text-primary;
        color: $color;
        &:hover,
        &:visited {
          color: $color;
        }
      }

      &:last-child {
        display: inline-block;
        text-overflow: ellipsis;
        overflow: hidden;
      }
      &:not(:last-child) {
        &::after {
          $circle-size: 4px;
          content: "";
          display: inline-block;
          width: $circle-size;
          height: $circle-size;
          border-radius: 50%;
          margin: $spacing-lg;
          background-color: $color-text-disabled;
        }
      }
    }
  }
}
