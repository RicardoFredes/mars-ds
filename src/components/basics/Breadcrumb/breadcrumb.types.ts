import type { DropdownMenuItemProps } from "../DropdownMenuItem";
import type { HTMLProps } from "react";

export interface BreadcrumbProps extends Omit<HTMLProps<HTMLDivElement>, "list"> {
  list?: DropdownMenuItemProps[];
}
