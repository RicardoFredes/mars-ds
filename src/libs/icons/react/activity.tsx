/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconActivity = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <polyline points="22 12 18 12 15 21 9 3 6 12 2 12" />
</svg>
);
  
export default IconActivity;