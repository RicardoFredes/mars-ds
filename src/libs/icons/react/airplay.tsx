/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAirplay = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  stroke-linejoin="round"
>
  <path d="M5 17H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-1" />
  <polygon points="12 15 17 21 7 21 12 15" />
</svg>
);
  
export default IconAirplay;