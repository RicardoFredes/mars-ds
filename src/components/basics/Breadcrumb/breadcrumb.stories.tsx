import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Breadcrumb from "./";
import { mockBreadcrumbListExpand, mockDefaultBreadcrumbs } from "./breadcrumb.fixture";

export default {
  title: "Components/Breadcrumb",
  component: Breadcrumb,
  argTypes: {},
} as ComponentMeta<typeof Breadcrumb>;

const Template: ComponentStory<typeof Breadcrumb> = (args) => <Breadcrumb {...args} />;

const DefaultArgs = {
  list: mockDefaultBreadcrumbs,
};

export const Default = Template.bind({});
Default.args = DefaultArgs;

export const Expanded = Template.bind({});
Expanded.args = {
  list: mockBreadcrumbListExpand,
};
