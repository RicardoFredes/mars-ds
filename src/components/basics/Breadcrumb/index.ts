/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 */

import Breadcrumb from "./breadcrumb.component";
export default Breadcrumb;
export * from "./breadcrumb.types";
