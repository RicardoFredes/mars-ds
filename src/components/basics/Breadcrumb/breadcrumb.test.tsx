import type { BreadcrumbProps } from "./breadcrumb.types";

import { fireEvent, render, waitFor } from "@testing-library/react";

import Breadcrumb from ".";
import { mountListMock } from "./breadcrumb.fixture";

const makeSut = (props?: Partial<BreadcrumbProps>) => render(<Breadcrumb {...props} />);

const PROPS = {
  ITEM_TEST_ID: "breadcrumb-item",
  TOGGLE_TEST_ID: "toggle-dropdown-button",
};

describe("<Breadcrumb>", () => {
  describe("when has less than 4 items", () => {
    const labels = ["Label 1", "Label 2", "Label 3"];
    const list = mountListMock(labels);

    it("should match the snapshot", () => {
      const wrapper = makeSut({ list });
      expect(wrapper).toMatchSnapshot();
    });

    it.each(labels)("should render a <BreadcrumbListItem> with label %s", (label) => {
      const wrapper = makeSut({ list });
      wrapper.getByText(label);
    });
  });

  describe("when has more than 3 items", () => {
    const labels = ["Label 1", "Label 2", "Label 3", "Label 4"];
    const list = mountListMock(labels);

    describe("When modal is closed", () => {
      it("should match the snapshot", () => {
        const wrapper = makeSut({ list });
        expect(wrapper).toMatchSnapshot();
      });

      it("should render a first <BreadcrumbListItem> with label 1", () => {
        const wrapper = makeSut({ list });
        wrapper.getByText(labels[0] as string);
      });

      it("should render a last <BreadcrumbListItem> with label 4", () => {
        const wrapper = makeSut({ list });
        wrapper.getByText(labels[labels.length - 1] as string);
      });

      it("should render a <ToggleDropdown>", () => {
        const wrapper = makeSut({ list });
        wrapper.getByTestId(PROPS.TOGGLE_TEST_ID);
      });

      it("should only render the first and the last <BreadcrumbListItem>", async () => {
        const wrapper = makeSut({ list });
        expect(wrapper.getAllByTestId(PROPS.ITEM_TEST_ID)).toHaveLength(2);

        await waitFor(() => {
          expect(wrapper.queryByText(labels[0] as string)).toBeInTheDocument();
          expect(wrapper.queryByText(labels[1] as string)).not.toBeInTheDocument();
          expect(wrapper.queryByText(labels[2] as string)).not.toBeInTheDocument();
          expect(wrapper.queryByText(labels[3] as string)).toBeInTheDocument();
        });
      });
    });

    describe("When modal is open", () => {
      it("should match the snapshot", () => {
        const wrapper = makeSut({ list });
        fireEvent.click(wrapper.getByTestId(PROPS.TOGGLE_TEST_ID));
        expect(wrapper).toMatchSnapshot();
      });

      it.each(labels)("should render a <BreadcrumbListItem> with label %s", (label) => {
        const wrapper = makeSut({ list });
        fireEvent.click(wrapper.getByTestId(PROPS.TOGGLE_TEST_ID));
        wrapper.getByText(label);
      });
    });
  });
});
