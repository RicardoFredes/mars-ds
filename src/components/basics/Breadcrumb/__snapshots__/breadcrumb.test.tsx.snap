// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb> when has less than 4 items should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumbs"
        class="breadcrumb"
      >
        <ol>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-0"
            >
              Label 1
            </a>
          </li>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-1"
            >
              Label 2
            </a>
          </li>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-2"
            >
              Label 3
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ol>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-0"
          >
            Label 1
          </a>
        </li>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-1"
          >
            Label 2
          </a>
        </li>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-2"
          >
            Label 3
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Breadcrumb> when has more than 3 items When modal is closed should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumbs"
        class="breadcrumb"
      >
        <ol>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-0"
            >
              Label 1
            </a>
          </li>
          <li>
            <div
              class="toggle-dropdown"
            >
              <button
                class="btn btn--neutral btn--size-sm toggle-btn toggle-btn--variant-text"
                data-testid="toggle-dropdown-button"
                title="Abrir mais opções para o item"
                type="button"
              >
                <span
                  class="icon btn__icon"
                  data-testid="more-horizontal"
                >
                  <svg
                    fill="none"
                    height="18"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
                      fill="currentColor"
                    />
                    <path
                      d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
                      fill="currentColor"
                    />
                    <path
                      d="M5 14C6.10457 14 7 13.1046 7 12C7 10.8954 6.10457 10 5 10C3.89543 10 3 10.8954 3 12C3 13.1046 3.89543 14 5 14Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </li>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-3"
            >
              Label 4
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ol>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-0"
          >
            Label 1
          </a>
        </li>
        <li>
          <div
            class="toggle-dropdown"
          >
            <button
              class="btn btn--neutral btn--size-sm toggle-btn toggle-btn--variant-text"
              data-testid="toggle-dropdown-button"
              title="Abrir mais opções para o item"
              type="button"
            >
              <span
                class="icon btn__icon"
                data-testid="more-horizontal"
              >
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
                    fill="currentColor"
                  />
                  <path
                    d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
                    fill="currentColor"
                  />
                  <path
                    d="M5 14C6.10457 14 7 13.1046 7 12C7 10.8954 6.10457 10 5 10C3.89543 10 3 10.8954 3 12C3 13.1046 3.89543 14 5 14Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-3"
          >
            Label 4
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Breadcrumb> when has more than 3 items When modal is open should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumbs"
        class="breadcrumb"
      >
        <ol>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-0"
            >
              Label 1
            </a>
          </li>
          <li>
            <div
              class="toggle-dropdown toggle-dropdown--is-open"
            >
              <button
                class="btn btn--neutral btn--size-sm btn--is-pressed toggle-btn toggle-btn--variant-text"
                data-testid="toggle-dropdown-button"
                title="Abrir mais opções para o item"
                type="button"
              >
                <span
                  class="icon btn__icon"
                  data-testid="more-horizontal"
                >
                  <svg
                    fill="none"
                    height="18"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
                      fill="currentColor"
                    />
                    <path
                      d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
                      fill="currentColor"
                    />
                    <path
                      d="M5 14C6.10457 14 7 13.1046 7 12C7 10.8954 6.10457 10 5 10C3.89543 10 3 10.8954 3 12C3 13.1046 3.89543 14 5 14Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
              <div>
                <div
                  class="card card--elevation-hg dropdown-menu toggle-dropdown__list"
                  data-testid="toggle-dropdown-list"
                >
                  <ul
                    class="dropdown-menu__list"
                  >
                    <li
                      class="dropdown-menu__list-item"
                    >
                      <a
                        class="dropdown-menu-item"
                        href="#item-1"
                        tabindex="0"
                      >
                        <span
                          class="text text--size-sm dropdown-menu-item__text"
                        >
                          Label 2
                        </span>
                      </a>
                    </li>
                    <li
                      class="dropdown-menu__list-item"
                    >
                      <a
                        class="dropdown-menu-item"
                        href="#item-2"
                        tabindex="0"
                      >
                        <span
                          class="text text--size-sm dropdown-menu-item__text"
                        >
                          Label 3
                        </span>
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li
            class="breadcrumb__item"
            data-testid="breadcrumb-item"
          >
            <a
              class="subtitle subtitle--size-md"
              href="#item-3"
            >
              Label 4
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ol>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-0"
          >
            Label 1
          </a>
        </li>
        <li>
          <div
            class="toggle-dropdown toggle-dropdown--is-open"
          >
            <button
              class="btn btn--neutral btn--size-sm btn--is-pressed toggle-btn toggle-btn--variant-text"
              data-testid="toggle-dropdown-button"
              title="Abrir mais opções para o item"
              type="button"
            >
              <span
                class="icon btn__icon"
                data-testid="more-horizontal"
              >
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 14C13.1046 14 14 13.1046 14 12C14 10.8954 13.1046 10 12 10C10.8954 10 10 10.8954 10 12C10 13.1046 10.8954 14 12 14Z"
                    fill="currentColor"
                  />
                  <path
                    d="M19 14C20.1046 14 21 13.1046 21 12C21 10.8954 20.1046 10 19 10C17.8954 10 17 10.8954 17 12C17 13.1046 17.8954 14 19 14Z"
                    fill="currentColor"
                  />
                  <path
                    d="M5 14C6.10457 14 7 13.1046 7 12C7 10.8954 6.10457 10 5 10C3.89543 10 3 10.8954 3 12C3 13.1046 3.89543 14 5 14Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
            <div>
              <div
                class="card card--elevation-hg dropdown-menu toggle-dropdown__list"
                data-testid="toggle-dropdown-list"
              >
                <ul
                  class="dropdown-menu__list"
                >
                  <li
                    class="dropdown-menu__list-item"
                  >
                    <a
                      class="dropdown-menu-item"
                      href="#item-1"
                      tabindex="0"
                    >
                      <span
                        class="text text--size-sm dropdown-menu-item__text"
                      >
                        Label 2
                      </span>
                    </a>
                  </li>
                  <li
                    class="dropdown-menu__list-item"
                  >
                    <a
                      class="dropdown-menu-item"
                      href="#item-2"
                      tabindex="0"
                    >
                      <span
                        class="text text--size-sm dropdown-menu-item__text"
                      >
                        Label 3
                      </span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </li>
        <li
          class="breadcrumb__item"
          data-testid="breadcrumb-item"
        >
          <a
            class="subtitle subtitle--size-md"
            href="#item-3"
          >
            Label 4
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
